runbotics-orchestrator/src/main/java/com/runbotics/service/impl/GuestServiceImpl.java

```diff
@@ -39,7 +39,9 @@ public GuestServiceImpl(
     }
 
     public boolean verifyGuestLimit(String guestIp) {
-        Optional<Guest> guestOptional = guestRepository.findById(guestIp);
+        Optional<Guest> guestOptional = guestRepository.findAll().stream()
+            .filter(guest -> passwordEncoder.matches(guestIp, guest.getIp()))
+            .findFirst();
```

Comment:
It looks good for me. I'm just wondering if there is another way to find just one guest user or it makes no difference?

Reply:
The reason why I did it this way is that bcrypt generates a unique hash every time, so we are not able to make such a query to the database that will check if the guest with that ip_hash already exists